drivers/sensor/kx022/kx022.c

```diff
@@ -0,0 +1,881 @@
+/* Kionix KX022 3-axis accelerometer driver
+ *
+ * Copyright (c) 2021 G-Technologies Sdn. Bhd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT kionix_kx022
+
+#include <drivers/sensor.h>
+#include <kernel.h>
+#include <device.h>
+#include <init.h>
+#include <string.h>
+#include <sys/byteorder.h>
+#include <sys/__assert.h>
+#include <logging/log.h>
+
+#include "kx022.h"
+
+LOG_MODULE_REGISTER(KX022, CONFIG_SENSOR_LOG_LEVEL);
+ static struct kx022_data dev_data;
+static struct kx022_config dev_config = {
+	.comm_master_dev_name = DT_INST_BUS_LABEL(0),
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(spi)
+#error "KX022 : KX022 SPI NOT IMPLEMENTED"
+#elif DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
+	.bus_init = kx022_i2c_init,
+#else
+#error "KX022 : BUS MACRO NOT DEFINED IN DTS"
+#endif
+
+#ifdef CONFIG_KX022_TRIGGER
+	.irq_port = DT_INST_GPIO_LABEL(0, int_gpios),
+	.irq_pin = DT_INST_GPIO_PIN(0, int_gpios),
+	.irq_flags = DT_INST_GPIO_FLAGS(0, int_gpios),
+#endif
+
+};
+
+//#define kx022_delay(msec) k_msleep(((uint16_t)msec) * 1000 / CONFIG_KX022_ODR + 1);
+
+#ifdef CONFIG_KX022_DIAGNOSTIC_MODE
+static struct kx022_configuration kx022_setting;
```

Comment:
Please move this struct inside `struct kx022_data`, so that it will be a bit easier when we convert this driver to support multi instance